node/src/components/storage.rs

```diff
@@ -434,22 +434,21 @@ impl Storage {
         Ok(match req {
             StorageRequest::PutBlock { block, responder } => {
                 let mut txn = self.env.begin_rw_txn()?;
-                if !txn.put_value(
-                    self.block_body_db,
-                    block.header().body_hash(),
-                    block.body(),
-                    true,
-                )? {
+                if !txn.put_value(self.block_body_db, &block.body().hash(), block.body(), true)? {
                     error!("Could not insert block body for block: {}", block);
                     txn.abort();
                     return Ok(responder.respond(false).ignore());
                 }
-                if !txn.put_value(self.block_header_db, block.hash(), block.header(), true)? {
+                if !txn.put_value(
+                    self.block_header_db,
+                    &block.header().hash(),
```

Comment:
(This is the same as `block.hash()`, is it? You just changed it because it's more natural to write it this way when storing the header?)

Reply:
This way relies less on the invariant that `block.header.body_hash == block.body.hash()`. It may be worth writing a check here for that invariant.